xds/internal/xdsclient/xds.go

```diff
@@ -943,19 +964,41 @@ func securityConfigFromCommonTLSContextUsingNewFields(common *v3tlspb.CommonTlsC
 	//  - certificate_name
 	//    -  this is an opaque name passed to the certificate provider
 	var validationCtx *v3tlspb.CertificateValidationContext
-	switch common.GetValidationContextType().(type) {
+	switch typ := common.GetValidationContextType().(type) {
 	case *v3tlspb.CommonTlsContext_ValidationContext:
 		validationCtx = common.GetValidationContext()
 	case *v3tlspb.CommonTlsContext_CombinedValidationContext:
 		validationCtx = common.GetCombinedValidationContext().GetDefaultValidationContext()
 	case nil:
 		// It is valid for the validation context to be nil on the server side.
 		return sc, nil
-	}
-	if validationCtx == nil || validationCtx.GetCaCertificateProviderInstance() == nil {
-		// Bail out if the `CertificateProviderPluginInstance` message is not
-		// found through one of the way detailed above.
-		return nil, nil
+	default:
+		return nil, fmt.Errorf("validation context contains unexpected type: %T", typ)
+	}
+	// If we get here, it means that the `CertificateValidationContext` message
+	// was found through one of the supported ways. It is an error if the
+	// validation context is specified, but it does not contain the
+	// ca_certificate_provider_instance field which contains information about
+	// the certificate provider to be used for the root certificates.
+	if validationCtx.GetCaCertificateProviderInstance() == nil {
+		return nil, fmt.Errorf("expected field ca_certificate_provider_instance is missing in CommonTlsContext message: %+v", common)
+	}
+	// The following fields are ignored:
+	// - trusted_ca
+	// - watched_directory
+	// - allow_expired_certificate
+	// - trust_chain_verification
+	switch {
+	case len(validationCtx.GetVerifyCertificateSpki()) != 0:
```

Comment:
How are those unsupported fields picked?

Are they "the remaining fields other than those we support"?
Or they are manually picked?
Will there be new fields that we don't support, and want to NACK?

Reply:
The decision on which fields to support and which to ignore was made during the gRFC review. The rationale behind the decision was this:
```
An unsupported field is normally ignored. However if ignoring a field compromises security, 
or if the unsupported field affects how we interpret other fields, we NACK the update when
the field is present.
```

Usually, in other messages, we only care about the fields which we support. But in the security configuration, we had to clearly make that distinction as to which of the unsupported fields to ignore and which of those to NACK.